syntax = "proto3";

message Request {
    map<string, string> kwargs = 1;
}

message Response {
    map<string, string> response = 1;
}

service Agent {
    rpc InitializeAgents(Request) returns (Response) {}
    rpc HandleMessage(Request) returns (Response) {}
    rpc InvokeSoloAgent(Request) returns (Response) {}
    rpc GetAgents(Request) returns (Response) {}
    rpc SetAgentFields(Request) returns (Response) {}
    rpc HandleSlashCommand(Request) returns (Response) {} 
    rpc HandleUserUpdate(Request) returns (Response) {}
    rpc HandleMessageReaction(Request) returns (Response) {}
}